Updated oauth sso documentation (#141)
slingr-stack/slingr-docs: content/en/dev-reference/security/single-sign-on.md

  <br>

  ---

## **OAuth**

> OAuth is an open-standard authorization protocol that allows secure third-party access to a user's resources without sharing their credentials (see [OAuth 2.0](https://en.wikipedia.org/wiki/OAuth)).

Reviewer: Let's use the official site of Oauth
```suggestion
OAuth is an open-standard authorization protocol that allows secure third-party access to a user's resources without sharing their credentials (see [OAuth 2.0](https://auth0.com/intro-to-iam/what-is-oauth-2)).
```

  const userDataResponse = svc.http.post({
    url: 'https://shopify.com/87175397693/account/customer/api/2024-10/graphql',
    params: {
      query: 'query { customer { id firstName lastName emailAddress {emailAddress} } }'
    },
    headers: {
      'Content-Type': 'application/json',
      Authorization : accessTokenResponse.access_token
    }
  });
  if (!userDataResponse || !userDataResponse.data || !userDataResponse.data.customer) throw('Error obtaining customer data')
  // fill out userRecord
  let email;
  if (userDataResponse.data.customer.emailAddress) {
    email = userDataResponse.data.customer.emailAddress.emailAddress;
    if (email) {
      const userRegistered = sys.data.findOne('sys.users', {email: email});
      if (userRegistered) {
        return userRegistered;
      }
    }
  }
  const firstName = userDataResponse.data.customer.firstName || 'Default first name';
  const lastName = userDataResponse.data.customer.lastName || 'Default last name';
  if (!email) throw('Missing required fields: email') // you might throw an exception for firstName and lastName
  userRecord.field('firstName').val(firstName);
  userRecord.field('lastName').val(lastName)

>   userRecord.field('email').val(email)

Reviewer: I would add an optional field to indicate that it is a good time to update the user.
```suggestion
  userRecord.field('firstName').val(firstName);                              // Required
  userRecord.field('lastName').val(lastName);                              // Required
  userRecord.field('email').val(email);                                           // Required
  userRecord.field('currentlyWorkingOn').val("Google");              // Optional
```

>   return sys.data.save(userRecord);

Reviewer: I would explicitly indicate that the userRecord must be returned.
```suggestion
  sys.data.save(userRecord);
  return userRecord;
```
